### PR TITLE
[metrics] Document the semantics of each Prometheus metric we use in the dashboard

### PR DESCRIPTION
**Pull request by ericl**
Signed-off-by: Eric Liang <ekhliang@gmail.com>

## Why are these changes needed?

This adds a short explanation of each metric we have in the dashboard. I kept it a little high level so we don't get too tied to particular label semantics, which we may change in the future.

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
LGTM!

File: doc/source/ray-observability/ray-metrics.rst [hunk 1]
```diff
@@ -75,16 +75,99 @@ Then go to to the location of the binary and run grafana using the built in conf
     ./bin/grafana-server --config /tmp/ray/session_latest/metrics/grafana/grafana.ini web
 
 Now, you can access grafana using the default grafana url, `http://localhost:3000`.
-If this is your first time, you can login with the username: `admin` and password `admin`.
+You can then see the default dashboard by going to dashboards -> manage -> Ray -> Default Dashboard. The same metric graphs are also accessible via :ref:`Ray Dashboard <ray-dashboard>`.
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-metrics.rst:78]: Can you have a direct link to the metrics section?

File: doc/source/ray-observability/ray-metrics.rst [hunk 2]
```diff
@@ -75,16 +75,99 @@ Then go to to the location of the binary and run grafana using the built in conf
...
+
+.. list-table:: Ray System Metrics
+   :header-rows: 1
+
+   * - Prometheus Metric
+     - Labels
+     - Description
+   * - `ray_tasks`
+     - `Name`, `State`
+     - Current number of tasks (both remote functions and actor calls) by state. The State label (e.g., RUNNING, FINISHED, FAILED) describes the state of the task. See rpc::TaskState for more information. The function/method name is available as the Name label.
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-metrics.rst:105]: Is it possible to have a code link to `rpc::TaskState`?

File: doc/source/ray-observability/ray-metrics.rst [hunk 3]
```diff
@@ -75,16 +75,99 @@ Then go to to the location of the binary and run grafana using the built in conf
...
+   * - `ray_actors`
+     - `Name`, `State`
+     - Current number of actors in a particular state. The State label is described by rpc::ActorTableData proto in gcs.proto. The actor class name is available in the Name label.
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-metrics.rst:108]: Same as above comment!

File: doc/source/ray-observability/ray-metrics.rst [hunk 4]
```diff
@@ -75,16 +75,99 @@ Then go to to the location of the binary and run grafana using the built in conf
...
+   * - `ray_resources`
+     - `Name`, `State`
+     - Logical resource usage for each node of the cluster. Each resource has some quantity that is in USED state vs AVAILABLE state. The Name label defines the resource name (e.g., CPU, GPU).
+   * - `ray_object_store_memory`
+     - `Location`, `ObjectState`
+     - Object store memory usage in bytes, broken down by logical Location (SPILLED, IN_MEMORY, etc.), and ObjectState (UNSEALED, SEALED).
```
Comment (rkooo567) [inline doc/source/ray-observability/ray-metrics.rst:114]: Is the definition of them documented somewhere? I'd be nice if we have explanation for each state (SPILLED vs IN_MEMORY).

**Comment by rkooo567** [comment]
Btw, one concern I have is the doc becomes outdated as time goes. Maybe we can have a comment at the grafana generation code to update this doc? Or we can even auto generate this, but I am not sure how easy that will be

**Comment by ericl** [comment]
Good point--- added the comment as suggested.

**Review by maxpumperla — APPROVED** [review]

**Pull request merged**